src/components/product/Gallery/Gallery.js

```diff
@@ -0,0 +1,60 @@
+import { useState } from "react";
+import { createGalleryAdapter } from "./adapters/image-gallery";
+import {
+  GalleryContainer,
+  Prev,
+  Next,
+  CurrentImage,
+  GridImage,
+  Image,
+} from "./styled";
+import PropTypes from "prop-types";
+
+const Gallery = ({ gallery }) => {
+  const images = createGalleryAdapter(gallery);
+  const [currentImage, setCurrentImage] = useState(0);
+  let max = images.length - 1 ?? 0;
+
+  const handleMainImage = (url) => {
```

Comment:
why not just setCurrentImage?

Reply:
I was looking for make more reusable this function but no happen, is a good idea only use the setter